utils/parametric-components.stanza

```diff
@@ -0,0 +1,537 @@
+            (_) :
+              yield(myk => myv)
+    ; println("param transformed %_ into %_" % [params replaced-params])
+    replaced-params
+      
+  
+  defn resolve-components (my-para:HashTable<Double|Toleranced, Vector<JITXObject>>,
+                           my-fix:HashTable<Double|Toleranced, Vector<JITXObject>>) ->  Vector<Instantiable> :
+    inside pcb-module :
+      val new-comps = Vector<Instantiable>()
+      for rs in my-para do :
+        val r-val = key(rs)
+        for my-inst in value(rs) do :
+          if not has-property?(my-inst.user-params) :
+            fatal("Instance %_ does not have expected property (user-params)" % [ref(my-inst)])
+          val user-params = replace-param(property(my-inst.user-params))
+          ; println("Searching with user-params => %_" % [user-params])
+          val new-comp =
+            if has-property?(my-inst.resistor) :
+              try :
+                chip-resistor(user-params)
+              catch (e):
+                println("Warning: generic resistor substitution for %_" % [ref(my-inst)])
+                match(property(my-inst.resistance)) :
+                  (r:Double) : dummy-resistor(r)
+                  (t:Toleranced) : dummy-resistor(find-std-val(t), t)
+            else if has-property?(my-inst.capacitor) :
+              try :
+                ceramic-cap(user-params)
+              catch (e):
+                println("Warning: generic capacitor substitution for %_" % [ref(my-inst)])
+                match(property(my-inst.capacitance)) :
+                  (c:Double) : dummy-capacitor(c)
+                  (t:Toleranced) : dummy-capacitor(find-std-val(t), t)
+            else if has-property?(my-inst.inductor) :
+              try :
+                smd-inductor(user-params)
+              catch (e):
+                println("Warning: generic inductor substitution for %_" % [ref(my-inst)])
+                match(property(my-inst.inductance)) :
+                  (l:Double) : dummy-inductor(l)
+                  (t:Toleranced) : dummy-inductor(find-std-val(t), t)
+            else :
+              fatal("unknown component found")
+          inst my-new-comp : new-comp
+          property(my-new-comp.name) = to-symbol(ref(my-inst))
+          ; hook in new component to old parametric instance pins (in parallel)
+          net (my-new-comp.p[1] my-inst.p[1])
+          net (my-new-comp.p[2] my-inst.p[2])
+          add(new-comps, new-comp)
+    new-comps
+
+  within transform-module(m) :
+    inside pcb-module :
+      val knownR = HashTable<Double|Toleranced, Vector<JITXObject>>()
+      val knownC = HashTable<Double|Toleranced, Vector<JITXObject>>()
+      val knownL = HashTable<Double|Toleranced, Vector<JITXObject>>()
+      val paraR = HashTable<Double|Toleranced, Vector<JITXObject>>()
+      val paraC = HashTable<Double|Toleranced, Vector<JITXObject>>()
+      val paraL = HashTable<Double|Toleranced, Vector<JITXObject>>()
```

Comment:
You can create them with `HashTable-init<Double|Toleranced, Vector<JITXObject>>(Vector<JITXObject>{})`.

This way `table[k]` will initialize the value to an empty vector first if needed.